cmd/skaffold/app/cmd/debug.go

```diff
@@ -21,26 +21,51 @@ import (
 	"io"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
 	debugging "github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy"
 )
 
 // NewCmdDebug describes the CLI command to run a pipeline in debug mode.
+// Unlike `dev`, `debug` defaults `auto-build` and `auto-deploy` to `false`.
```

Comment:
I am not sure that same flag can have different values based on the command. Could you make sure it really works?